src/Tickets/Commerce/Order_Modifiers/Modifiers/Modifier_Abstract.php

```diff
@@ -147,33 +141,27 @@ public function update_modifier( array $data ): Model {
 		// Ensure the modifier_type is set to the expected one.
 		$data['modifier_type'] = $this->modifier_type;
 
-		try {
-			$this->validate_data( $data );
-		} catch ( InvalidArgumentException $exception ) {
-			new Order_Modifier( [] );
-		}
-
-		// Use the repository to update the data in the `order_modifiers` table.
+		$this->validate_data( $data );
 		return $this->repository->update( new Order_Modifier( $data ) );
 	}
 
 	/**
-	 * Retrieves the modifier by its ID.
+	 * Retrieves modifier data by ID.
 	 *
 	 * @since TBD
 	 *
 	 * @param int    $modifier_id The modifier ID.
-	 * @param string $modifier_type The modifier type.
+	 *
 	 *
 	 * @return array|null The modifier data or null if not found.
 	 */
-	public function get_modifier_by_id( int $modifier_id, string $modifier_type ): ?array {
+	public function get_modifier_by_id( int $modifier_id ): ?array {
 		$modifier_data = $this->repository->find_by_id( $modifier_id );
 		return $modifier_data ? $modifier_data->to_array() : null;
 	}
 
 	/**
-	 * Finds a modifier by its slug.
+	 * Finds a modifier by its display name.
```

Comment:
```suggestion
	 * Search modifiers by criteria.
```
I'm not sure of the exact text, but this doesn't search by just display name.

Reply:
Right now I think we can get away with just searching by display name. 

Looking at the Figma https://www.figma.com/design/BPWocsWvjojTaiip1ivW5A/Coupons-%2B-Fees?node-id=0-1&node-type=canvas&t=pyvfWQJNJFjcqbW6-0 we just have the search based off of the display name.